src/ioq_server2.erl

```diff
@@ -233,8 +248,10 @@ get_concurrency() ->
 
 -spec set_concurrency(non_neg_integer()) -> non_neg_integer().
 set_concurrency(C) when is_integer(C), C > 0 ->
-    lists:foldl(
-        fun(Pid, Total) ->
+    lists:foldl(fun
```

Comment:
I think you can avoid the extra clause if your would have specialized functions in the ioq_sup.

```erlang
ioq_sup:ioq2_servers() -> 
    ioq_sup:ioq2_servers_by_type(search) ++ ioq_sup:ioq2_servers_by_type(normal).

ioq_sup:ioq2_servers_by_type(search) 
ioq_sup:ioq2_servers_by_type(normal)
``` 

Then you would rewrite this function to use `ioq_sup:ioq2_servers_by_type(normal)`.

Reply:
I realized that this clause is unnecessary because we're using `ioq_sup:get_ioq_servers()` here which does not include the search pid.